eo-maven-plugin/src/main/java/org/eolang/maven/fp/Footprint.java

```diff
@@ -0,0 +1,198 @@
+            throw new IllegalStateException(
+                String.format("Source file %s does not exist", this.source)
+            );
+        }
+        final Path result;
+        if (Footprint.exists(this.target) && Footprint.isAfter(this.target, this.source)) {
+            result = this.target;
+        } else {
+            result = this.dependentOnCache(content);
+        }
+        return result;
+    }
+
+    /**
+     * Updates target depends on global cache.
+     * @param content Content function
+     * @return Path to target
+     * @throws IOException If fails to check files last modified time
+     */
+    private Path dependentOnCache(final Func<Path, String> content) throws Exception {
+        final Path result;
+        if (this.cache.cacheable()) {
+            final Path cached = this.cache.path();
+            if (Footprint.exists(cached) && Footprint.isAfter(cached, this.source)) {
+                result = this.copiedFromCache();
+            } else {
+                result = this.updatedAll(content);
+            }
+        } else {
+            result = this.updatedTarget(content);
+        }
+        return result;
+    }
+
+    /**
+     * Takes content given function and update target file.
+     * @param content Content function
+     * @return Absolute path to target file
+     */
+    private Path updatedTarget(final Func<Path, String> content) throws Exception {
+        return Footprint.updated(this.source, this.target, content);
+    }
+
+    /**
+     * Takes content from given function, updates target file, updates global cache file.
+     * @param content Content function
+     * @return Absolute path to target file
+     * @throws Exception If failed to extract content or update files
+     */
+    private Path updatedAll(final Func<Path, String> content) throws Exception {
+        final Func<Path, String> sticky = new StickyFunc<>(content);
+        final Path result = this.updatedTarget(sticky);
+        Footprint.updated(this.source, this.cache.path(), sticky);
+        return result;
+    }
+
+    /**
+     * Takes content from global cache and updates target file.
+     * @return Absolute path to target file
+     */
+    private Path copiedFromCache() throws Exception {
+        return Footprint.updated(
+            this.cache.path(),
+            this.target,
+            src -> new TextOf(src).asString()
+        );
+    }
+
+    /**
+     * Takes content from given function and updates given file.
+     * @param source Absolute path to source file
+     * @param dest Absolute path to file to update
+     * @param content Content function
+     * @return Absolute path to file to update
+     * @throws Exception If failed to extract content or update files
+     */
+    private static Path updated(
+        final Path source, final Path dest, final Func<Path, String> content
+    ) throws Exception {
+        new Saved.Default(content.apply(source), dest).value();
```

Comment:
@maxonfjvipon also, it looks a bit weird that you can't just create `Saved.Default` where you need it and let it "act" like `dest`. Here, you created a special static function to create `Saved.Default` and then call its `value()`. Looks like a mix of OOP and procedural style, for no obvious reason.